python/test_prereq_parse.py

```diff
@@ -176,3 +180,174 @@ def test_prereqs_coreqs(index=None, shorten=False, verbose=False, hard=False):
+                        "exprs": [
+                            {"type": "ATOM", "exprs": "CHEM 1570"},
+                            {"type": "ATOM", "exprs": "CHEM 3570"},
+                            {"type": "ATOM", "exprs": "CHEM 3580"},
+                        ],
+                    },
+                    {"type": "ATOM", "exprs": "FDSC 4170"},
+                ],
+            },
+        ),
+        (
+            "(MATH 2210 AND MATH 2220) OR (MATH 2230 AND MATH 2240) OR (MATH 1920 AND MATH 2940)",
+            {
+                "type": "OR",
+                "exprs": [
+                    {
+                        "type": "AND",
+                        "exprs": [
+                            {"type": "ATOM", "exprs": "MATH 2210"},
+                            {"type": "ATOM", "exprs": "MATH 2220"},
+                        ],
+                    },
+                    {
+                        "type": "AND",
+                        "exprs": [
+                            {"type": "ATOM", "exprs": "MATH 2230"},
+                            {"type": "ATOM", "exprs": "MATH 2240"},
+                        ],
+                    },
+                    {
+                        "type": "AND",
+                        "exprs": [
+                            {"type": "ATOM", "exprs": "MATH 1920"},
+                            {"type": "ATOM", "exprs": "MATH 2940"},
+                        ],
+                    },
+                ],
+            },
+        ),
+        (
+            "(PHYS 2208 AND (CHEM 2080 OR MATH 2090)) OR (MATH 2130 OR MATH 2310 OR MATH 2220)",
+            {
+                "type": "OR",
+                "exprs": [
+                    {
+                        "type": "AND",
+                        "exprs": [
+                            {"type": "ATOM", "exprs": "PHYS 2208"},
+                            {
+                                "type": "OR",
+                                "exprs": [
+                                    {"type": "ATOM", "exprs": "CHEM 2080"},
+                                    {"type": "ATOM", "exprs": "MATH 2090"},
+                                ],
+                            },
+                        ],
+                    },
+                    {
+                        "type": "OR",
+                        "exprs": [
+                            {"type": "ATOM", "exprs": "MATH 2130"},
+                            {"type": "ATOM", "exprs": "MATH 2310"},
+                            {"type": "ATOM", "exprs": "MATH 2220"},
+                        ],
+                    },
+                ],
+            },
+        ),
+    ]
+
+    correct = 0
+    total = len(test_cases)
+    for test in test_cases:
+        (boolean_string, answer) = test
+        try:
+            response = parse_boolean_string(boolean_string)
+        except:
+            print("PARSING ERROR")
+            continue
+        if response == answer:
```

Comment:
```suggestion
        if response == answer:
```
Can we do `assert(response == answer)`?